racket/collects/syntax/to-string.rkt

```diff
@@ -50,11 +50,12 @@
                                    (printf "; ")))
                            l))]
               [(and (pair? (syntax-e c))
-                    (eq? (syntax-e (car (syntax-e c))) 'quote))
+                    (eq? (syntax-e (car (syntax-e c))) 'quote)
+                    (eq? (syntax-span (car (syntax-e c))) 1))
```

Comment:
Could you add a comment that when the above three conditions are met, it must be the case that it's the shorthand form, which always constructs a syntax list, justifying the use of `syntax->list` below.